src/model.ts

```diff
@@ -132,6 +122,65 @@ class FileBrowserModel implements IDisposable {
   }
 
   /**
+   * Select an item by name.
+   *
+   * #### Notes
+   * This is a no-op if the name is not valid or already selected.
+   */
+  select(name: string): void {
+    if (name in this._selection && !this._selection[name]) {
+      this._selection[name] = true;
+      this.selectionChanged.emit(void 0);
+    }
+  }
+
+  /**
+   * De-select an item by name.
+   *
+   * #### Notes
+   * This is a no-op if the name is not valid or not selected.
+   */
+  deselect(name: string): void {
+    if (name in this._selection && this._selection[name]) {
+      this._selection[name] = false;
+      this.selectionChanged.emit(void 0);
+    }
+  }
+
+  /**
+   * Check whether an item is selected.
+   *
+   * #### Notes
+   * Returns `true` for a valid name that is selected, `false` otherwise.
+   */
+  isSelected(name: string): boolean {
+    return !!this._selection[name];
+  }
+
+  /**
+   * Get the list of selected names.
+   */
+  getSelected(): string[] {
+    let output: string[] = [];
+    for (let key in this._selection) {
+      if (this._selection[key]) {
+        output.push(key);
+      }
+    }
+    return output;
+  }
+
+  /**
+   * Clear the selected items.
+   */
+  clearSelected(): void {
```

Comment:
Can't do that, I rely on all of the names being in there, how about this?

`Object.keys(this._selection).map(k => this._selection[k] = false);`